compiler/rustc_passes/src/liveness.rs

```diff
@@ -1590,6 +1594,26 @@ impl<'tcx> Liveness<'_, 'tcx> {
                             .emit();
                     },
                 )
+            } else if can_remove {
+                self.ir.tcx.struct_span_lint_hir(
+                    lint::builtin::UNUSED_VARIABLES,
+                    first_hir_id,
+                    hir_ids_and_spans.iter().map(|(_, pat_span, _)| *pat_span).collect::<Vec<_>>(),
+                    |lint| {
+                        let mut err = lint.build(&format!("unused variable: `{}`", name));
+                        err.multipart_suggestion(
+                            "try removing the field",
+                            hir_ids_and_spans
+                                .iter()
+                                .map(|(_, pat_span, _)| (*pat_span, String::new()))
+                                .collect(),
+                            // REVIEW: `pat_span` is too small to automatically remove, but I don't
+                            // see a clean way to fix it
```

Comment:
Have you tried [SourceMap::span_extend_to_next_char()](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_span/source_map/struct.SourceMap.html#method.span_extend_to_next_char)? You should be able to get the SourceMap via `self.ir.tcx.sess.source_map()`.